File: backend/app/schemas/metrics.py
"""
Pydantic schemas endpoint Fase 7 — cost-summary & inventory-metrics
(docs/ARCHITECTURE.md §5).
"""
from decimal import Decimal

from pydantic import BaseModel


class CostSummaryOut(BaseModel):
    run_id: str
    total_ordering_cost: Decimal
    total_holding_cost: Decimal
    total_inventory_cost: Decimal  # usulan ForecastIQ
    baseline_inventory_cost: Decimal  # existing (planning)
    savings_pct: Decimal


class InventoryMetricOut(BaseModel):
    target_type: str  # product / material
    target_id: str
    scope: str  # baseline / forecastiq
    service_level: Decimal
    fill_rate: Decimal
    stock_out_rate: Decimal
    inventory_turnover: Decimal

File: frontend/src/types/metrics.ts
// Cocok dengan backend/app/schemas/metrics.py (Fase 7).
// Nilai Decimal diserialisasi backend sebagai string (envelope AGENTS.md §4).

export interface CostSummary {
  run_id: string;
  total_ordering_cost: string;
  total_holding_cost: string;
  total_inventory_cost: string; // usulan ForecastIQ
  baseline_inventory_cost: string; // existing (planning perusahaan)
  savings_pct: string;
}

export type MetricScope = "baseline" | "forecastiq";

export interface InventoryMetric {
  target_type: string; // product | material
  target_id: string;
  scope: MetricScope;
  service_level: string;
  fill_rate: string;
  stock_out_rate: string;
  inventory_turnover: string;
}
